graph/schema.resolvers.go

```diff
@@ -0,0 +1,213 @@
+	var abilityList []*model.PokemonAbility
+	err := context.Database.Find(&abilityList).Error
+	if err != nil {
+		return nil, err
+	}
+	currentId, errConvertId := getCurrentId(abilityList)
+	if errConvertId != nil {
+		return nil, err
+	}
+	newAbility := &model.PokemonAbility{
+		ID:   strconv.Itoa(currentId + 1),
+		Name: input.Name,
+	}
+	err2 := context.Database.Create(&newAbility).Error
+	if err2 != nil {
+		return nil, err2
+	}
+	return newAbility, nil
+}
+
+// CreateCategory is the resolver for the createCategory field.
+func (r *mutationResolver) CreateCategory(ctx context.Context, input model.NewCategory) (*model.PokemonCategory, error) {
+	context := GetContext(ctx)
+	var categoryList []*model.PokemonCategory
+	err := context.Database.Find(&categoryList).Error
+	if err != nil {
+		return nil, err
+	}
+	currentId, errConvertId := getCurrentId(categoryList)
+	if errConvertId != nil {
+		return nil, err
+	}
+	newCategory := &model.PokemonCategory{
+		ID:   strconv.Itoa(currentId + 1),
+		Name: input.Name,
+	}
+	err2 := context.Database.Create(&newCategory).Error
+	if err2 != nil {
+		return nil, err2
+	}
+	return newCategory, nil
+}
+
+// ListPokemons is the resolver for the listPokemons field.
+func (r *queryResolver) ListPokemons(ctx context.Context) ([]*model.Pokemon, error) {
+	context := GetContext(ctx)
+	var pokemonList []*model.Pokemon
+	err := context.Database.Preload("Type").Preload("Category").Preload("Ability").Find(&pokemonList).Error
+	if err != nil {
+		return nil, err
+	}
+	return pokemonList, nil
+}
+
+// ListType is the resolver for the listType field.
+func (r *queryResolver) ListType(ctx context.Context) ([]*model.PokemonType, error) {
+	context := GetContext(ctx)
+	var typeList []*model.PokemonType
+	err := context.Database.Find(&typeList).Error
+	if err != nil {
+		return nil, err
+	}
+	return typeList, nil
+}
+
+// ListCategory is the resolver for the listCategory field.
+func (r *queryResolver) ListCategory(ctx context.Context) ([]*model.PokemonCategory, error) {
+	context := GetContext(ctx)
+	var categoryList []*model.PokemonCategory
+	err := context.Database.Find(&categoryList).Error
+	if err != nil {
+		return nil, err
+	}
+	return categoryList, nil
+}
+
+// ListAbility is the resolver for the listAbility field.
+func (r *queryResolver) ListAbility(ctx context.Context) ([]*model.PokemonAbility, error) {
+	context := GetContext(ctx)
+	var abilityList []*model.PokemonAbility
+	err := context.Database.Find(&abilityList).Error
+	if err != nil {
+		return nil, err
+	}
+	return abilityList, nil
+}
+
+// FindPokemonByID is the resolver for the findPokemonById field.
+func (r *queryResolver) FindPokemonByID(ctx context.Context, id string) (*model.Pokemon, error) {
+	context := GetContext(ctx)
+	var pokemon *model.Pokemon
+	err := context.Database.Preload("Type").Preload("Category").Preload("Ability").Where("id = ?", id).Find(&pokemon).Error //if cannot find pokemon return null % show error data not found
```

Comment:
fix the comment

Reply:
Oh This comment , i remember that you told me a long timeeee  to handle the error & the way to respond to it, i will delete it naka.